### ISSUE TITLE
Remove hard-coded logging

### ISSUE THREAD

**Issue by ivanjaros** [stephenafamo/goldmark-pdf#11]
If there is error during rendering, like missing image, I will get a log:
```
2023/01/11 16:45:04 IMAGE ERROR: foobar, file does not exist
```

This is not good because I have no control over this. I don't even think this should be a log. It should be simply returned error which can be ignored, or not. Adding something like WithLog(someInterace) may sound like good idea at first but in reality it is useless. Hence I think there should dedicated error that can join multiple errors(multiple missing images) which can be then processed by the caller or ignored, depending on severity(ie var ErrMissingImage = errors.New(...) with errors.Is()). The final document can still be used even with missing images hence why it should not be a hard error. Also, this specifically ties to https://github.com/stephenafamo/goldmark-pdf/issues/7

**Comment by stephenafamo** [comment]
I'll just wait untill `Go 1.20` releases in the next few months which allow wrapping of multiple errors.

The downside is that it will then require any users to upgrade to `1.20`, but everything should be backwards compatible so no issues there.

Then it'll be easy to ignore missing image errors with something like


```go
if err != nil && !errors.Is(err, pdf.ErrMissingImage) {
    // ....
}
```


EDIT: That may not work, because that will suppress every error if **any** error is the `pdf.ErrMissingImage`

**Comment by ivanjaros** [comment]
That's what unwrapping of errors is for. That is of no concern. I think that missing images should not even cause errors to begin with. As the alt issue i linked, the document should render alt text or some placeholder.

**Comment by stephenafamo** [comment]
I agree it shouldn't prevent rendering and should use the alt text instead.

Still, a user should have a way to know if such a thing happens during rendering, for example, a browser will show a console error and an error in the network tab.

Maybe a middle ground will be an option to indicate whether not-found images should be treated as errors or not

**Comment by ivanjaros** [comment]
True, but you are rendering essentially an html page where missing images are perfectly fine and the sole reason alt even exists. If this would be a dedicated pdf builder/renderer, sure, but it is not. It's literally markdown renderer.
